tests/tests_pytorch/loops/test_loops.py

```diff
@@ -844,9 +844,9 @@ def _get_iterator(self):
             # iterable check
             0,
             # epoch ends
+            0,
             1,
             # teardown
-            1,
```

Comment:
Is this not getting called on `teardown` anymore because `_load_current_iterator` will set `self.iterators = []` and trigger garbage collection and thus iterator shutdown?

Reply:
Yeah! Should be that `__next__` gets called after the last item in the last dataloader -> raises `StopIteration` -> catch and trigger `_use_next_iterator` -> set `self.iterators = []` -> workers get shutdown -> raise `StopIteration` -> Lightning now knows that we ran out of data in the epoch